.github/workflows/tests.yml

```diff
@@ -1,46 +1,222 @@
-      fail-fast: true
       matrix:
+        # Tests must be run on all target platforms and Python versions
+        os: [ubuntu-latest, macos-latest, windows-latest]
         python-version: [3.6, 3.7, 3.8]
-
+      # Do not cancel in-progress jobs if any matrix job fails
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
     steps:
-    - uses: actions/checkout@v2
+    - name: Checkout repository
+      uses: actions/checkout@v2
     - name: Set up Python ${{ matrix.python-version }}
       uses: actions/setup-python@v2
       with:
         python-version: ${{ matrix.python-version }}
-    - name: Install dependencies
+    - name: Export ${HOME}/.local/bin to ${PATH}
+      # Executable Python binaries are ususally stored there
+      run: echo "${HOME}/.local/bin" >> ${GITHUB_PATH}
+    - name: Get pip cache dir
+      # pip's cache path depends on the operating system, see
+      # https://github.com/actions/cache/blob/main/examples.md#python---pip
+      # This requires pip >=20.1
+      id: pip-cache
       run: |
-        python -m pip install --upgrade pip
-        python -m pip install flake8 flake8-docstrings pytest
-        python -m pip install numpy
-        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
-    - name: Lint code with flake8
+        python -m pip install --user --upgrade pip
+        echo "::set-output name=dir::$(pip cache dir)"
+    - name: Create/Restore Cache
+      uses: actions/cache@v2
+      with:
+        path: |
+          ${{ steps.pip-cache.outputs.dir }}/**
+          ./docs/build/**
+          ~/docs/build/**
+          ./.pytest_cache/**
+          ~/.pytest_cache/**
+        key: ${{ runner.os }}-${{ matrix.python-version }}-${{ github.job }}
+        restore-keys: |
+          ${{ runner.os }}-${{ matrix.python-version }}-
+          ${{ runner.os }}-
+    - name: Install/Upgrade setuptools and wheel
+      run: python -m pip install --user --upgrade setuptools wheel
+    - name: Test installation of MDTools
       run: |
-        # stop the build if there are Python syntax errors or undefined names
-        flake8 . --count --select=E1,E2,E3,E4,E9,F63,F4,F7,F82,W1,W2,W3 --show-source --statistics
-        # exit-zero treats all errors as warnings
-        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=79 --statistics
-    - name: Lint docstrings with flake8
+        python -m pip install --user --upgrade .
+        python -c "import mdtools"
+    - name: Install/Upgrade requirements to build the documentation
+      run: python -m pip install --user --upgrade -r docs/requirements.txt
+    - name: Test build of the documentation
+      run: make -C ./docs/ html
+    - name: doctest
+      # Because the default dtypes of NumPy arrays are different on
+      # Windows, doctest will detect false failures. See e.g.
+      # https://stackoverflow.com/questions/36278590/numpy-array-dtype-is-coming-as-int32-by-default-in-a-windows-10-64-bit-machine
```

Comment:
Just out of curiosity: Which tests fail on windows? Would those also fail an any other 32 bit arch?

Reply:
All test were numpy arrays are printed fail, because the dtype of the array differs from what doctest is expecting. doctest is a quite stupid testing method. It already fails if a white space is not at the location that doctest is expecting it to be.

It could well be that doctest detects failures on any other 32 bit architectures as well. But due to the above mentioned reason, you shouldn't rate the doctest output to high. If pytest passes, everything should be fine (of course, we need a complete test suite for pytest).

Reply:
It's unfortunate that doctests tests the output in such a strict way. If it's only the tests and nothing under laying it's sensible to skip windows.